app/pdf_utils/image_handler.py

```diff
@@ -0,0 +1,144 @@
+    "subscript": lambda e: f"\\textsubscript{{{e}}}",
+    # use the `\ul` command instead of the `\underline` as
+    # `\underline` "wraps" the argument in a horizontal box
+    # which doesn't allow for linebreaks
+    "underline": lambda e: f"\\ul{{{e}}}",
+    "italic": lambda e: f"\\textit{{{e}}}",
+    "bold": lambda e: f"\\textbf{{{e}}}",
+}
+
+
+def wrap_caption_text(data: Dict) -> NoEscape:
+    """
+    Wraps text item with Latex commands corresponding to the sibbling elements
+    of the text item. Allows for wrapping multiple formatting options.
+    ----
+    eg: if data looks like: {"bold": true, "italic": true, "text": "text to format" }
+    then `wrap_caption_text(data)` will return: `\\bold{\\italic{text to format}}`
+
+    """
+    # pull caption text
+    caption_text = data["text"]
+
+    # escape caption text
+    caption_text = escape_latex(caption_text)
+
+    # process command for each option if found in caption textleaf (data)
+    for option, command in TEXT_WRAPPERS.items():
+
+        # TODO ensure this logic is applying correctly
+        if (data[option] is True) and (caption_text.strip(" ") != ""):
+
+            # run the command for the data attributes that are true
+            caption_text = command(caption_text)
+
+    return NoEscape(caption_text)
+
+
+PLACEHOLDER_OBJECT_KEY = "No Image Found"
+PLACEHOLDER_CAPTION = "No Caption Provided"
+
+
+def process_image_caption(caption_text_leaf: Dict) -> List[NoEscape]:
+    """
+    Uses a modified text processing logic to process image captions
+    """
+
+    # pass to wraptext/caption format util
+    processed_caption = NoEscape(wrap_caption_text(caption_text_leaf))
+
+    return processed_caption
+
+
+def process_image(
+    data: Dict,
+    atbd_id: int = None,
+):
+    """
+    Top level processing of each of the possible section items: subsection, image,
+    table, generic div-type node, ordered and unordered list, and equation.
+
+    Returns Latex formatted object for each of these, to be appeneded to the
+    Latex document.
+    """
+
+    # data is a Dict
+    for _indx, element in enumerate(data["children"]):  # type: ignore
+        # source for type ignore: https://github.com/python/mypy/issues/2220
+        # data['children']: List
+        # _indx: int
+        # element: Dict
+
+        # if content type is img
+        if element["type"] == "img":
+
+            # get the image block info: objectKey
+            # maintain camelCase
+            objectKey = pydash.get(
+                obj=element, path="objectKey", default=PLACEHOLDER_OBJECT_KEY
+            )
+
+        # get the image block info: caption
+        if element["type"] == "caption":
+
+            # get the caption text leaf
+            caption_text_leaf: Dict = pydash.get(
+                obj=element, path="children.0", default=PLACEHOLDER_CAPTION
+            )
+
+            # handle errors and try to provide useful feedback during image load failure
+            try:
```

Comment:
I'm certainly not a python expert, so feel free to disregard this comment, but I might try to avoid the nested try-catch statements - I would handle all of the possible exceptions within the first try-catch block. Or perhaps set some sort of error flag within the exception, if you have more logic to execute before raising the exception based on flag